### PR TITLE
[WIP] feat(roblox): luau string interpolation

### LINKED ISSUE #235: Luau string interpolation
https://github.com/Roblox/luau/pull/614

### PR DESCRIPTION
**Pull request by sno2**
Closes #235

### CODE REVIEW

**Comment by Kampfkarren** [comment]
I'm surprised tests passed on this. Your formatting is wrong, please run cargo fmt (and ideally, set your editor to save on format)

**Comment by sno2** [comment]
I actually setup formatting on my editor before starting this PR. `cargo fmt` does the same thing as what I'm doing. I don't know why it looks weird in the macros, though.

**Comment by Kampfkarren** [comment]
I guess it just doesn't filter in those, please fix the indenting

File: full-moon/src/atom.rs [hunk 1]
```diff
@@ -46,6 +46,74 @@ fn read_string(lex: &mut Lexer<Atom>, quote: char) -> bool {
     false
 }
 
+#[inline]
```
Comment (Kampfkarren) [inline full-moon/src/atom.rs:49]: We shouldn't frivolously `#[inline]` stuff, remove the lot of it

**Comment by Kampfkarren** [comment]
FYI this might not be merged for a bit. Is this for Hacktoberfest? If so I can apply the label

**Comment by Kampfkarren** [comment]
Hey, I'm going to be working on this now in #253, thanks for the contribution though!

**Pull request closed without merging**